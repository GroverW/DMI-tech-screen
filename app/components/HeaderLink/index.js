/**
 * A generic anchor tag
 */

import styled from 'styled-components';
import { NavLink } from 'react-router-dom';

const HeaderLink = styled(NavLink)`
  color: ${props => props.theme.font};
  text-decoration: none;
  padding: 0px 5px;
  margin: 0px 10px;
  line-height: 24px;
  height: 30px;

  &:hover,
  &.active {
    border-bottom: 4px solid ${props => props.theme.buttonBorder};
  }
`;

export default HeaderLink;
