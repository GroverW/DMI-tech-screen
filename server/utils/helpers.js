const { RESULTS_PER_PAGE } = require('../config');

const getPageStart = page =>
  page && Number.isInteger(+page)
    ? Math.max(page - 1, 0) * RESULTS_PER_PAGE
    : 0;

module.exports = {
  getPageStart,
};
