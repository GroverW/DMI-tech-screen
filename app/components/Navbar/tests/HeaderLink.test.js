/**
 *
 * Tests for HeaderLink
 *
 * @see https://github.com/react-boilerplate/react-boilerplate/tree/master/docs/testing
 *
 */

import React from 'react';
import { renderWithRouter } from '../../../tests/helpers/testHelper';
// import 'jest-dom/extend-expect'; // add some helpful assertions

import HeaderLink from '../HeaderLink';

describe('<HeaderLink />', () => {
  it('Expect to not log errors in console', () => {
    const spy = jest.spyOn(global.console, 'error');
    renderWithRouter(<HeaderLink to="" />);
    expect(spy).not.toHaveBeenCalled();
  });

  it('Should render and match the snapshot', () => {
    const {
      container: { firstChild },
    } = renderWithRouter(<HeaderLink to="" />);
    expect(firstChild).toMatchSnapshot();
  });
});
