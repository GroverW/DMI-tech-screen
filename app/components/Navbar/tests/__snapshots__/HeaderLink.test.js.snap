// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HeaderLink /> Should render and match the snapshot 1`] = `
<a
  aria-current="page"
  class="HeaderLink-sc-16cp3q6-0 bczvcN active"
  href="/"
/>
`;
